Move Javascript in status.html to separate file

jakethakur opened the pull request:
Moves the javascript in status.html to a new status.js file.

Closes #123

jakethakur committed:
Move Javascript in status.html to separate file

manu-chroma commented:
Hey Jake. You can click on detail to check the build log for travis for your particular PR. Check it out here: https://travis-ci.org/manu-chroma/username-availability-checker/builds/449037018?utm_source=github_status&utm_medium=notification for the current failing build. 

Also, try testing your changes locally. That might help you in knowing what might have gone wrong (apart from the logs)

jakethakur committed:
Removed trailing whitespace

jakethakur commented:
OK, I believe it was just that I left some trailing whitespace (though the log is fairly verbose so I might have missed something else further down).

manu-chroma commented:
I kind of agree. The log went just too verbose on failure. Some of the initial stuff was not very useful as such. Maybe we can do something about that later.

manu-chroma reviewed (approved):
LGTM!

jakethakur commented:
Great, thank you for your help!

manu-chroma commented:
I will let someone else have a look too, before I merge this one!

kx-chen reviewed (commented):
Small issue: there are template tags (which would normally be rendered by flask) in the Javscript which do not get rendered when they are moved to a separate file. This causes the entire site to break. Not @jakethakur 's fault. Looking into this more. Other than that, it looks good to me!

kx-chen reviewed (approved):
Looks good to me!

andrewda commented on static/js/status.js:
```diff
@@ -0,0 +1,108 @@
+function gtag(){dataLayer.push(arguments);}
+gtag('js', new Date());
+
+gtag('config', 'UA-72356300-6');
```
Why is this being moved? I think this should probably stay where it was. It's best practice to keep this Google Analytics stuff in the `<script>` tag so that it can easily be updated later if the Analytics/gtag stuff ever updates.

jakethakur replied:
Sure, sorry about that! I'll move that back. I wasn't sure if you wanted me to move that code as well.

andrewda commented on static/js/status.js:
```diff
@@ -0,0 +1,108 @@
+		console.error("Incorrect logo key in yml.")
+	}
+	return attribute;
+}
```
Add a newline @ end of file

jakethakur committed:
Moved Google Analytics script back to status.html

jakethakur committed:
Fixed indentation

jakethakur commented:
OK, that should have fixed it :)

utkarsh2102 reviewed (approved):
Looks good! :D

manu-chroma commented:
We have to consider something like this: https://stackoverflow.com/questions/37259740/passing-variables-from-flask-to-javascript/37260465

or maybe a better solution for js file to have that config.

kx-chen commented:
I've got a solution that's similiar to @manu-chroma 's suggestion. My thoughts are that it should be addressed in a different issue/pr, as it is a bit out of the scope of this task.

manu-chroma committed:
chore: merge master and resolve conflicts

manu-chroma committed:
gitignore: exclude vscode/

manu-chroma committed:
feat: use data from backend in status.js

manu-chroma committed:
frontend.py: remove signup link support

manu-chroma committed:
status.js: remove signup support and misc improvements

manu-chroma committed:
fix: use css rather than js based fontawesome render

JS based rendering interferes with our own custom icons
which have similar name to font awesome classes resulting
in replacement with a not found svg.

ghost commented:
how to answer this question? im new
